controllers/app.controller.js

```diff
@@ -32,17 +34,40 @@ exports.getArticlesById = (req, res, next) => {
     .catch(next);
 };
 
-exports.getArticlesComments = (req, res, next) => {
+exports.getArticleComments = (req, res, next) => {
   const { article_id } = req.params;
-  const promises = [fetchArticlesComments(article_id)];
+  fetchArticlesById(article_id)
+    .then(() => {
+      return fetchArticleComments(article_id);
+    })
+    .then((comments) => {
+      res.status(200).send({ comments });
+    })
+    .catch(next);
+};
 
-  if (article_id) {
-    promises.push(fetchArticlesById(article_id));
+exports.postComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+
+  if (!username || !body) {
+    return next({
+      status: 400,
+      msg: "Bad Request",
+    });
   }
 
+  const promises = [
+    fetchArticlesById(article_id),
+    fetchUsersByUsername(username),
```

Comment:
This is not required, if you look at the seed for comments you'll see a foreign key constraint on users. If the user does not exist you'll get an error you can handle (which you have not tested for FYI)